eventually/authentication/models.py

```diff
@@ -0,0 +1,161 @@
+    middle_name = models.CharField(('middle name'), max_length=20)
+    last_name = models.CharField(('second name'), max_length=20)
+    email = models.EmailField(('email address'), max_length = 40, unique = True)
+    password = models.CharField(('password'), max_length = 128)
+    updated_at = models.DateTimeField(auto_now_add = True)
+    created_at = models.DateTimeField(auto_now_add = True, editable = False)
+
+    def __str__(self):
+        """
+            This magic method is redefined to show first, last name and id of the CustomUser object.
+        """
+        return '{} {} {}'.format(self.first_name, self.last_name, self.id)
+
+    @staticmethod
+    def get_by_id(user_id):
+        """
+        :param user_id: the id of a user to be found in the DB
+        :return: user object or None if a user with such ID does not exist
+        """
+        try:
+            return CustomUser.objects.get(id=user_id)
+        except CustomUser.DoesNotExist:
+            return None
+
+    @staticmethod
+    def delete_by_id(user_id):
+        """
+        :param user_id: an id of a user to be deleted
+        :return: True if object existed in the db and was removed or False if it didn't exist
+        """
+        try:
+            user = CustomUser.get_by_id(user_id)
+            user.delete()
+            return True
+        except AttributeError:
+            return False
+
+    @staticmethod
+    def create(first_name, middle_name, last_name, email, password):
+        """
+        :param first_name: first name of a user
+        :param middle_name: middle name of a user
+        :param last_name: last name of a user
+        :param email: email of a user
+        :param password: password of a user
+        :return: a new user object which is also written into the DB
+        """
+        user = CustomUser(first_name=first_name,
+                          last_name=last_name,
+                          middle_name=middle_name,
+                          email=email,
+                          password=password)
+        user.save()
+        return user
+
+    #'FIXME: remove 4 methods below? Do we need them?'
+    def get_full_name(self):
+        """
+        :return: first, middle and last name of the created user
+        """
+
+        return '{} {} {}'.format(self.first_name, self.middle_name, self.last_name)
+
+    def get_email(self):
+        """
+        :return: email of the user.
+        """
+        return self.email
+
+    def get_updated_at(self):
+        """
+        :return: returns last update date in form of a timestamp
+        """
+        date = format(self.updated_at, u'U')
+        return date
+
+    def get_created_at(self):
+        """
+        :return: creation date of a user record in form of a timestamp
+        """
+        return format(self.created_at, u'U')
+
+    #'FIXME: which of these to remove? to_dict or myattr_to_dict ?'
+    def to_dict(self):
```

Comment:
We have to create our custom `to_dict` method cause magic method `__dict__` can return irrelevant info.

Reply:
Agree. Magic method removed.